src/Angor/Client/Shared/NavMenuState.cs

```diff
@@ -1,9 +1,72 @@
-﻿namespace Angor.Client.Shared
+﻿using Angor.Client.Services;
+using Angor.Client.Storage;
+
+namespace Angor.Client.Shared
 {
-    public class NavMenuState 
+    public class NavMenuState
     {
+        private readonly ICacheStorage _cacheStorage;
+
+        public NavMenuState(ICacheStorage cacheStorage)
+        {
+            _cacheStorage = cacheStorage;
+        }
+
         public event Action OnChange;
 
+        private string _activePage = string.Empty;
+        public string ActivePage
+        {
+            get => _activePage;
+            private set
+            {
+                if (_activePage != value)
+                {
+                    _activePage = value;
+                    PersistState();
```

Comment:
hmm I am not sure how I fee about pushing to storage every time we change a page, do we need to do that? anyway each page will set itself when it is loaded, can you explain again why do we need to persist  the page?